### ISSUE TITLE
IBAN code -> stacktrace

### ISSUE THREAD

**Issue by qwertologe** [marytts/marytts#480]
Try to speak DE12345678901234567890 and MaryTTS will fail with a stacktrace.

This is critical because IBAN codes are often used.

The stacktrace is with version 5.1 but you can reproduce it in the demo (http://mary.dfki.de:59125/), too.

```
<html><body><h1>Internal server error</h1><p>Processing failed.<pre>java.lang.Exception: Module Preprocess: Problem processing the data.
        at marytts.server.Request.processOneChunk(Request.java:551)
        at marytts.server.Request.processOrLookupOneChunk(Request.java:389)
        at marytts.server.Request.process(Request.java:336)
        at marytts.server.http.SynthesisRequestHandler.process(SynthesisRequestHandler.java:272)
        at marytts.server.http.SynthesisRequestHandler.handleClientRequest(SynthesisRequestHandler.java:98)
        at marytts.server.http.BaseHttpRequestHandler.handle(BaseHttpRequestHandler.java:144)
        at org.apache.http.nio.protocol.BufferingHttpServiceHandler$RequestHandlerAdaptor.handle(BufferingHttpServiceHandler.java:189)
        at org.apache.http.nio.protocol.SimpleNHttpRequestHandler.handle(SimpleNHttpRequestHandler.java:51)
        at org.apache.http.nio.protocol.AsyncNHttpServiceHandler.processRequest(AsyncNHttpServiceHandler.java:453)
        at org.apache.http.nio.protocol.AsyncNHttpServiceHandler.requestReceived(AsyncNHttpServiceHandler.java:225)
        at org.apache.http.nio.protocol.BufferingHttpServiceHandler.requestReceived(BufferingHttpServiceHandler.java:127)
        at org.apache.http.impl.nio.DefaultNHttpServerConnection.consumeInput(DefaultNHttpServerConnection.java:161)
        at org.apache.http.impl.nio.DefaultServerIOEventDispatch.inputReady(DefaultServerIOEventDispatch.java:147)
        at org.apache.http.impl.nio.reactor.BaseIOReactor.readable(BaseIOReactor.java:161)
        at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvent(AbstractIOReactor.java:335)
        at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvents(AbstractIOReactor.java:315)
        at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:275)
        at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
        at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:542)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NumberFormatException: For input string: "12345678901234567890"
        at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
        at java.lang.Long.parseLong(Long.java:444)
        at java.lang.Long.valueOf(Long.java:513)
        at java.lang.Long.decode(Long.java:665)
        at marytts.language.de.preprocess.NumberEP.expandInteger(NumberEP.java:279)
        at marytts.language.de.preprocess.AbbrevEP.expandAbbrev(AbbrevEP.java:186)
        at marytts.language.de.preprocess.AbbrevEP.expandAbbrev(AbbrevEP.java:203)
        at marytts.language.de.preprocess.AbbrevEP.expand(AbbrevEP.java:127)
        at marytts.language.de.preprocess.ExpansionPattern.process(ExpansionPattern.java:275)
        at marytts.language.de.Preprocess.matchAndExpandPatterns(Preprocess.java:118)
        at marytts.language.de.Preprocess.process(Preprocess.java:69)
        at marytts.server.Request.processOneChunk(Request.java:549)
        ... 19 more
</pre></body></html>

**Comment by psibre** [comment]
@Rootex do you have any input on this?

**Comment by psibre** [comment]
Full details of valid IBAN formats: https://www.swift.com/sites/default/files/resources/swift_standards_ibanregistry.pdf

**Comment by psibre** [comment]
**TL;DR:** `~/[A-Z]{2}\d{2}\s?(((?i)[A-Z]\d){4}\s?){1,5}\d{0,4}/` might work

**Comment by attacomsian** [comment]
Me and @Rootex found a work around for this. It was related to `AbbrevEP.java`. It always tried to expand it as an integer and since we have a certain maximum size for the integer, it was failing. We put a condition on the integer expansion to only expand when under maximum allowed size. Otherwise, it should expand as digits. 

We just created a new pull request #486.
